RedisCaching/CacheContext.cs

```diff
@@ -40,13 +40,13 @@ internal void SetItem<TValue>(CacheItem<TValue> cacheItem)
             client.Database.StringSet(cacheItem.Id, convert.ToRedis(cacheItem), expiry, When.Always, CommandFlags.FireAndForget);
         }
 
-        internal Task SetItemAsync<TValue>(CacheItem<TValue> cacheItem)
+        internal async Task SetItemAsync<TValue>(CacheItem<TValue> cacheItem)

```

Comment:
@gregoks I don't see reason to make this function async (it was already awaitable).

Reply:
Added ConfigureAwait since we need it to be awaitable all the way.